requests/auth.py

```diff
@@ -63,19 +64,26 @@ class HTTPDigestAuth(AuthBase):
     def __init__(self, username, password):
         self.username = username
         self.password = password
-        self.last_nonce = ''
-        self.nonce_count = 0
-        self.chal = {}
-        self.pos = None
-        self.num_401_calls = 1
+        # Keep state in per-thread local storage
+        self.tl = threading.local()
```

Comment:
Not trying to shave any yaks here, but I'd really prefer 2 things.

1. That the thread local be internal only, e.g., prefixed with a `_`
2. That it has a better name that people can understand without having to read the entirety of the class definition (e.g, `_thread_local`)